nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -103,6 +104,18 @@ def __init__(self, cfg: DictConfig, trainer: Trainer, no_lm_init=True):
 
         super().__init__(cfg, trainer=trainer, no_lm_init=no_lm_init)
 
+        with open_dict(self.cfg):
+            self.cfg.precision = trainer.precision
+        # TODO: @maanug-nv consolidate into one attribute (requires lots of changes in subclasses)
+        self.torch_dtype = utils_funcs.torch_dtype_from_precision(self.cfg.precision)  # Mixed precision datatype
+        self.autocast_dtype = self.torch_dtype  # Mixed precision datatype
```

Comment:
@maanug-nv if O2 is false, autocast_dtype would be set as fp32
we dont want that, we want autocast dtype as the precision fp16/bf16

this would cause autocast issues